rocAL/source/readers/image/caffe_lmdb_record_reader.cpp

```diff
@@ -38,51 +39,71 @@ CaffeLMDBRecordReader::CaffeLMDBRecordReader()
     _current_file_size = 0;
     _loop = false;
     _shuffle = false;
-    _file_id = 0;
     _last_rec = false;
     _file_count_all_shards = 0;
 }
 
 unsigned CaffeLMDBRecordReader::count_items() {
-    if (_loop)
-        return _file_names.size();
-
-    int ret = ((int)_file_names.size() - _read_counter);
+    int ret = 0;
```

Comment:
please add new base class and move all common stuff to that class

Reply:
Done